lib/fireblocks/api/client.go

```diff
@@ -0,0 +1,123 @@
+package fireblocks
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+
+	"github.com/golang-jwt/jwt"
+)
+
+type Client struct {
+	apiKey       string
+	clientSecret string
+	host         string
+	http         http.Client
+}
+
+const TransactionEndpoint string = "v1/transactions"
+
+func NewClient(apiKey string, clientSecret string, host string) *Client {
+	return &Client{
+		apiKey:       apiKey,       // pragma: allowlist secret
+		clientSecret: clientSecret, // pragma: allowlist secret
+		host:         host,
+	}
+}
+
+func (c *Client) sendRequest(ctx context.Context, req *http.Request, endpoint string, request []byte, nonce string) ([]byte, error) {
+	bodyHash := sha256.Sum256(request)
+	authToken, err := genJWTToken(endpoint, nonce, c.apiKey, string(bodyHash[:]), c.clientSecret)
+	if err != nil {
+		return nil, errors.Wrap(err, "genJWTToken")
+	}
+	req.Header = getHeader(authToken, c.apiKey)
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return nil, errors.New("http Do")
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			log.Error(ctx, "FireBlocks: closing body failure", err)
+		}
+	}()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "read response body")
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return body, errors.Wrap(err, "http response code: %d", resp.StatusCode)
+	}
+
+	return body, nil
+}
+
+func (c *Client) createPostRequest(ctx context.Context, endpoint string, request []byte) (*http.Request, error) {
+	endpoint = fmt.Sprintf("%s/%s", c.host, endpoint)
+	req, err := http.NewRequestWithContext(
+		ctx,
+		http.MethodPost,
+		endpoint,
+		bytes.NewReader(request),
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "new post request")
+	}
+
+	return req, nil
+}
+
+func (c *Client) createGetRequest(ctx context.Context, endpoint string) (*http.Request, error) {
```

Comment:
the only different between this function and createPostRequest is the method. Why not have a single function that takes the method as a parameter?